packages/react-components/src/components/BaseECharts/BaseECharts.tsx

```diff
@@ -66,31 +43,105 @@ export const BaseECharts = ({
     },
     styles,
   });
-
+  const optionRef = useRef({
+    ...DEFAULT_ECHART_OPTIONS,
+  });
+  const tooltipAddMode = useRef(false);
   const series = useMemo(
     () => dataStreamsToEChartSeries(dataStreams as DataStream[], chartType),
     [dataStreams, chartType]
   );
 
-  const option = useMemo(
-    () => ({
-      ...DEFAULT_ECHART_OPTIONS,
-      xAxis: { ...DEFAULT_X_AXIS, show: axis?.showX ?? true },
+  useEffect(() => {
+    let chart: EChartsType;
+    if (chartRef.current !== null) {
+      chart = init(chartRef.current);
+    }
+    return () => {
+      chart?.dispose();
+    };
+  }, []);
+
+  useEffect(() => {
+    let chart: EChartsType | undefined;
+    if (chartRef.current !== null) {
+      chart = getInstanceByDom(chartRef?.current);
+      chart?.resize({ width: size?.width, height: size?.height });
+    }
+  }, [size]);
+
+  useEffect(() => {
+    let chart;
+    optionRef.current = {
+      ...optionRef.current,
+      xAxis: { ...optionRef.current.xAxis, show: axis?.showX ?? true },
       yAxis: {
-        ...DEFAULT_Y_AXIS,
+        ...optionRef.current.yAxis,
         show: axis?.showY ?? true,
         min: yMin,
         max: yMax,
       },
-      title: {
-        text: series.length === 0 ? `${chartType} chart - no data present` : '',
-      },
-      series,
-    }),
-    [series, chartType, yMin, yMax, axis]
-  );
+    };
+    if (chartRef.current !== null) {
+      chart = getInstanceByDom(chartRef.current);
+      chart?.setOption(optionRef.current);
+    }
+  }, [yMin, yMax, axis, chartType]);
+
+  useEffect(() => {
+    let chart;
+    if (chartRef.current !== null) {
+      chart = getInstanceByDom(chartRef.current);
+      optionRef.current = {
+        ...optionRef.current,
+        graphic: updatePositions(optionRef.current, series, chart),
+        title: {
+          text: series.length === 0 ? `${chartType} chart - no data present` : `${chartType} chart`,
+        },
+        series,
+      };
+
+      chart?.setOption(optionRef.current);
+    }
+  }, [series]);
+
+  useEffect(() => {
+    let chart: EChartsType | undefined;
+    if (chartRef.current !== null) {
+      chart = getInstanceByDom(chartRef.current);
+      chart?.on('click', (e) => {
+        if (tooltipAddMode.current) {
+          const length = (optionRef.current.graphic as GraphicInternal[])?.length ?? 0;
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          optionRef.current?.graphic?.push(getToolTipGraphic(e, length, chart, size));
```

Comment:
what if we maintain state in the component for scooters? like a scooters useState list. we can convert this into a graphics list with all the other props we convert and call setOption in 1 place

Reply:
we can also make sure the positions are correct in the same conversion step.

Reply:
Also, I think it would be helpful to have a data-structure which represents the scooter, and then we convert that into the data-structure the chart requires to visualize it. I think that will make the code easier to reason about in the future in the case where we add another feature which also utilizes graphics

Reply:
yes, this is just a basic implementation, a lot of specific implementation details will change... wanted to have something in place so that we are working off of one file / base pattern, this is not the final solution